GitUI/UserControls/RevisionGrid/Graph/LaneNodeLocator.cs

```diff
@@ -22,40 +22,42 @@ public LaneNodeLocator(ILaneRowProvider laneRowProvider)
             {
                 // as unlikely as it may be...
                 // don't throw, just pretend we couldn't find it
-                return (null, false);
             }
-
-            var lane = x / laneWidth;
-            lock (_laneRowProvider)
+            else
             {
-                var laneRow = _laneRowProvider.GetLaneRow(rowIndex);
-                if (laneRow == null || lane >= laneRow.Count)
-                {
-                    return (null, false);
-                }
-
-                if (lane == laneRow.NodeLane)
-                {
-                    return (laneRow.Node, isAtNode: true);
-                }
-
-                for (int laneInfoIndex = 0, laneInfoCount = laneRow.LaneInfoCount(lane); laneInfoIndex < laneInfoCount; ++laneInfoIndex)
+                var lane = x / laneWidth;
+                lock (_laneRowProvider)
                 {
-                    // search for next node below this row
-                    var laneInfo = laneRow[lane, laneInfoIndex];
-                    var firstJunction = laneInfo.Junctions.First();
-                    for (int nodeIndex = 0, nodeCount = firstJunction.NodeCount; nodeIndex < nodeCount; ++nodeIndex)
+                    var laneRow = _laneRowProvider.GetLaneRow(rowIndex);
+                    if (laneRow != null)
                     {
-                        var laneNode = firstJunction[nodeIndex];
-                        if (laneNode.Index > rowIndex)
+                        if (lane == laneRow.NodeLane)
+                        {
+                            return (laneRow.Node, isAtNode: true);
+                        }
+
+                        if (lane < laneRow.Count)
                         {
-                            return (laneNode, isAtNode: false);
+                            for (int laneInfoIndex = 0, laneInfoCount = laneRow.LaneInfoCount(lane); laneInfoIndex < laneInfoCount; ++laneInfoIndex)
+                            {
+                                // search for next node below this row
+                                var laneInfo = laneRow[lane, laneInfoIndex];
+                                var firstJunction = laneInfo.Junctions.First();
+                                for (int nodeIndex = 0, nodeCount = firstJunction.NodeCount; nodeIndex < nodeCount; ++nodeIndex)
+                                {
+                                    var laneNode = firstJunction[nodeIndex];
+                                    if (laneNode.Index > rowIndex)
+                                    {
+                                        return (laneNode, isAtNode: false);
+                                    }
+                                }
+                            }
                         }
                     }
                 }
-
-                return (null, false);
             }
+
+            return (null, false);
         }
     }
 }
```

Comment:
Nitpick: missing NewLine at EOF

Reply:
@RussKie, older source files end with a NewLine, newer ones do not. What is the preferred style?